client.go

```
@@ -242,19 +242,42 @@ func (c *Client) fullURL(suffix string, args ...any) string {
 }
 
 func (c *Client) handleErrorResp(resp *http.Response) error {
-	var errRes ErrorResponse
-	err := json.NewDecoder(resp.Body).Decode(&errRes)
-	if err != nil || errRes.Error == nil {
-		reqErr := &RequestError{
+	contentType := resp.Header.Get("Content-Type")
+	mainContentType := strings.Split(contentType, ";")[0]
```

Should we verify here that splitting the contentType header does infact return results - otherwise we could gracefully handle that. I don't think it's safe to assume the response will contain the header. With that in mind, I also don't think we should return an error if the header doesn't exist - we should probably continue assuming that the response is infact json which is what we are doing now.

---

Yeah looks like the content type is an optional header, so I've reworked this to check for a content type, and if so, check if it's `text/plain` - if not, we will go the JSON route.